Lesson20/Merve/script.js

```diff
@@ -0,0 +1,127 @@
+// 1. Create a Person class with name, age, and gender properties, and a greet() method that logs a greeting message to the console.
+// Then create a Student class that extends the Person class and adds a major property. Override the greet() method in the Student class to log a different greeting message to the console. Create 2 variables, one of type Person and one of type Student and log them to console.
+class Person{
+    constructor(name,age,gender){
+        this._name = name;
+        this._age = age;
+        this._gender = gender;
+    }
+    greet(){
+        return `Hello, my name is ${this._name}`;
+    }
+}
+
+class Student extends Person{
+    constructor(name,age,gender,major){
+        super(name,age,gender);
+        this._major = major;
+    }
+    greet(){
+        return `Hello, my name is ${this._name} and i am ${this._age} ${this._gender}`;
+    }
+}
+const person1 = new Person('Sam', 40, 'male');
+const student1 = new Student('Emma', 38, 'female', 'computer science');
+
+console.log(person1, person1.greet());
+console.log(student1, student1.greet());
+// 2. Create a Shape class with name and color properties, and a describe() method that logs a description of the shape to the console (e.g. "This is a blue square"). Then create a Square class that extends the Shape class and adds sideLength and area properties. Override the describe() method in the Square class to log a description of the square to the console (e.g. "This is a blue square with a side length of 5 and an area of 25").
+class Shape{
+    constructor(name, color){
+        this._name = name;
+        this._color = color;
+    }
+    describe(){
+        return `This is a ${this._name} ${this._color}`;
+    }
+}
+
+class Square extends Shape{
+    constructor(name, color, sideLength, area){
```

Comment:
No need to pass area prop. We can always calculate the area of the square if we know the side length.

Reply:
```suggestion
    constructor(name, color, sideLength){
```